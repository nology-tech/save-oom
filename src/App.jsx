import { Route, Routes } from "react-router-dom";
import "./styles/main.scss";
import { UserProvider } from "./contexts/UserContext";
import Home from "./pages/Home/Home";
const App = () => {
  return (
    <UserProvider>
    <Routes>
      <Route path="*" element={<Home />} />
    </Routes>
    </UserProvider>
  )
};

export default App;


