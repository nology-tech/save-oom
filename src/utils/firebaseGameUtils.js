import { db } from "../firebase";
import {
  addDoc,
  collection,
  getDocs,
  query,
  where,
  orderBy,
  Timestamp,
} from "firebase/firestore";
/** 
 * Get all the rounds of a specific game and level for a given user, ordered by most recent first 
 @param {*} userId
  @param {String} game
*@returns a querySnap (document snapshot), returns a Promise.
*/

export const getGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** 
   * Get all the correct round of a specific game and level for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns a promise with the query snapshot.
  */
export const getCorrectGameRoundsForUser = async (id, game, level) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("level", "==", level),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect rounds for a given user, ordered by most recent first 
  @param {String} userId
  *@returns  a promise with the query snapshot.
  */
export const getIncorrectGameRoundsForUser = async (id) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("correct", "==", false),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect round of a specific game for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns  a promise with the query snapshot.
  */

export const getAllIncorrectGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};



/** Function that returns an array of rounds 
  @param {String} userId
  @param {String} gameId
  @param {Function} getGameRounds
  *@returns returs a promise.
  */
export const getArrayOfRounds = async (id, game, level, getGameRounds) => {
  const querySnap = await getGameRounds(id, game, level);
  const phonicsArr = querySnap.docs.map((doc) => {
    return doc.data().phonic;
  });
  return phonicsArr;
};

/** Function that returns an array of rounds 
  @param {String} id
  @param {String} game
  @param {Object} roundInfo
  @param {String} answer
  *@returns returs a promise.
  */

export const saveUserRound = async (id, game, level, gameStats, answer) => {
  const { isCorrect, score } = gameStats;
  await addDoc(collection(db, "users", id, "rounds_played"), {
    correct: isCorrect,
    gameId: game,
    phonic: answer,
    level: level,
    roundTime: Timestamp.fromDate(new Date()).toDate(),
    score: score,
  });
};
