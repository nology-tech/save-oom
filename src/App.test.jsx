import App from "./App";
import { customRender } from "./utils/testUtils";

it("Should render the app", () => {
  const { container } = customRender(<App />);

  expect(container).toMatchSnapshot();
});
