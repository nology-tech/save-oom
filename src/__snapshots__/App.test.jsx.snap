// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the app 1`] = `
<div>
  <div
    class="layout"
  >
    <h1>
      Save Oom!!
    </h1>
    <p>
      Let's save the little critter!
    </p>
    <div
      class="avatar-heading"
    >
      <h1
        class="avatar-heading__text"
      >
        Please add an avatar to save Oom
      </h1>
    </div>
    <div
      class="avatar-heading"
    >
      <h1
        class="avatar-heading__text"
      >
        Name your avatar
      </h1>
    </div>
    <div
      class="avatar-heading"
    >
      <h1
        class="avatar-heading__text"
      >
        Jack's a great name, Oom would be proud!
      </h1>
    </div>
  </div>
  <div
    class="nav-bar"
  >
    <div
      class="nav-bar__section"
    >
      <img
        alt="Logo image"
        class="logo__image"
        src="logo.png"
      />
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Phonics stages
        </h2>
        <img
          alt="Phonics stages"
          class="nav-bar-header__img"
          src="Vectorvector.png"
        />
      </div>
      <h2>
        Introduction
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 1
        </h2>
        <img
          alt="Level 1"
          class="nav-bar-header__img"
          src="Vectorlock.png"
        />
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 2
        </h2>
        <img
          alt="Level 2"
          class="nav-bar-header__img"
          src="Vectorlock.png"
        />
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Level 3
        </h2>
        <img
          alt="Level 3"
          class="nav-bar-header__img"
          src="Vectorlock.png"
        />
      </div>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Stats
        </h2>
        <img
          alt="Stats"
          class="nav-bar-header__img"
          src="Vectorstat.png"
        />
      </div>
      <h2>
        Jack
      </h2>
      <div
        class="nav-bar-header"
      >
        <h2
          class="nav-bar-header__heading"
        >
          Add Avatar
        </h2>
        <img
          alt="Add Avatar"
          class="nav-bar-header__img"
          src="Vectoradd.png"
        />
      </div>
    </div>
    <button
      class="button-logout"
    >
      Logout
    </button>
  </div>
</div>
`;
